CMakeLists.txt

```diff
@@ -221,6 +221,13 @@ else()
   add_definitions(-Wall -Wextra -pedantic -Wno-unused-parameter
     -Wstrict-prototypes -std=gnu99)
 
+  if(CMAKE_COMPILER_IS_GNUCC)
```

Comment:
Added detection of `-Wimplicit-fallthrough` so we can explicitly add it to the compile flags, and changed the attribute macro to be defined based on `#if __has_attribute(fallthrough)`.